### PR TITLE
packaging, tests: stop services in prerm

### PR DESCRIPTION
**Pull request by bboozzoo**
Fix package removal when there's snaps with services installed. Details are available in this forum topic: https://forum.snapcraft.io/t/stop-commands-and-snapd-package-cleanup/13688
In short, the `stop-command` is executed via `/usr/bin/snap run <stop-command>`, what implies that the `snap` command must be available at the time. Therefore, stopping services in postrm is too late, since the `snapd` package is gone.

### CODE REVIEW

File: tests/main/postrm-purge/task.yaml [hunk 1]
```diff
@@ -36,3 +48,14 @@ execute: |
             exit 1
         fi
     done
+
+    if [[ "$SPREAD_SYSTEM" != ubuntu-14.04-* ]]; then
+        # ubuntu-14.04: systemctl dos not list not-found & failed units properly
```
Comment (stolowski) [inline tests/main/postrm-purge/task.yaml]:
```suggestion
        # ubuntu-14.04: systemctl does not list not-found & failed units properly
```

File: packaging/ubuntu-14.04/snapd.prerm [hunk 2]
```diff
@@ -2,7 +2,40 @@
 
 set -e
 
+systemctl_stop() {
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply packaging/ubuntu-14.04/snapd.prerm]: I think we'd need to somehow generate those files before snapd deb is built. Hopefully the changes in this PR are one-off and we won't have to update it in a longer while.

File: tests/main/postrm-purge/task.yaml [hunk 3]
```diff
@@ -36,3 +48,14 @@ execute: |
             exit 1
         fi
     done
+
+    if [[ "$SPREAD_SYSTEM" != ubuntu-14.04-* ]]; then
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply tests/main/postrm-purge/task.yaml]: We're using bash pattern matching extension, with works with `==` and `!=` if enclosed by `[[ .. ]]`.

File: packaging/ubuntu-14.04/snapd.prerm (hunk 2 above)
[the remark this answers is not in the document]
Comment (anonymouse64) [reply packaging/ubuntu-14.04/snapd.prerm]: Okay, it's probably fine until we have to update it again eventually ... :man_shrugging:

File: tests/main/postrm-purge/task.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (anonymouse64) [reply tests/main/postrm-purge/task.yaml]: ah got it thanks

File: packaging/ubuntu-14.04/snapd.prerm (hunk 2 above)
[the remark this answers is not in the document]
Comment (zyga) [reply packaging/ubuntu-14.04/snapd.prerm]: I would love to see them generated. I agree that it's "probably okay" except when it won't be and we realise we have a skew.

**Pull request merged**